enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationRestService.java

```
@@ -207,24 +244,84 @@ public ObjectResponse<IdentityVerificationStatusResponse> checkIdentityVerificat
+        final var operationDescription = "submitting documents for verification";
 
-        final String operationDescription = "submitting documents for verification";
         checkApiAuthentication(apiAuthentication, operationDescription);
         checkEncryptionContext(encryptionContext, operationDescription);
         checkRequestObject(request, operationDescription);
 
-        // Extract user ID from onboarding process for current activation
+        final var requestObject = request.getRequestObject();
+        final var documentsByFilename = getDocumentsByFilename(encryptionContext.getActivationId(), requestObject);
+
+        final var documentsV2 = buildDocumentsV2(requestObject, documentsByFilename);
+
+        final var requestV2 = DocumentSubmitV2Request.builder()
+                .processId(requestObject.getProcessId())
+                .documents(documentsV2)
+                .resubmit(requestObject.isResubmit())
+                .build();
+
+        return submitDocuments(requestV2, encryptionContext);
+    }
+
+    private Map<String, Document> getDocumentsByFilename(final String activationId, final DocumentSubmitRequest request) {
+        final var documentsByFilename = new HashMap<String, Document>();
+
+        if (request.getData() != null) {
+            try {
+                final var extractedDocuments = dataExtractionService.extractDocuments(request.getData())
+                        .stream()
+                        .collect(Collectors.toMap(Document::getFilename, document -> document));
+
+                documentsByFilename.putAll(extractedDocuments);
+            } catch (final DocumentVerificationException e) {
+                logger.error("Unable to extract documents from {}, activationId: {}", request, activationId);
```

The exception is swallowed.

---

Yes, this is behavior of original [implementation](https://github.com/wultra/enrollment-server/blob/develop/enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentProcessingService.java#L490). It returns empty collection when exception is thrown and I didn't want to change it. However, I should add error message to the log and log the entire exception on debug level.

---

Yes, logging is enough.